web-common/src/lib/time/new-grains.ts

```diff
@@ -0,0 +1,367 @@
+import { V1TimeGrain } from "@rilldata/web-common/runtime-client";
+import type { DateTimeUnit } from "luxon";
+
+type TimeGrainAlias =
+  | "ms"
+  | "MS"
+  | "s"
+  | "S"
+  | "m"
+  | "h"
+  | "H"
+  | "d"
+  | "D"
+  | "w"
+  | "W"
+  | "M"
+  | "q"
+  | "Q"
+  | "y"
+  | "Y";
+
+export const grainAliasRegex = /(ms|s|S|m|h|H|d|D|w|W|M|q|Q|y|Y)/;
```

Comment:
missing capital MS which is present in TimeRangeAlias, either add at both places or remove from both ?